### PR TITLE
Implements deleting message

### PR DESCRIPTION
**Pull request by saketkumar95**
Fixes : https://github.com/zulip/zulip-mobile/issues/883

Used editing message api to update the message as `(deleted)` same as it is done on web.

### CODE REVIEW

File: src/i18n/translations/messages_en.json [hunk 1]
```diff
@@ -60,4 +60,5 @@
   "Forgot password?": "Forgot password?",
   "Members": "Members",
   "Group details": "Group details",
+  "Delete Message": "Delete Message",
```
Comment (borisyankov) [inline src/i18n/translations/messages_en.json]: Use the 'Delete message' casing.

File: src/message/messageActionSheet.js [hunk 2]
```diff
@@ -89,6 +90,10 @@ const editMessage = async ({ message, actions }: MessageAuthAndActions) => {
   actions.startEditMessage(message.id);
 };
 
+const deleteMessage = async ({ auth, message }: AuthAndMessageType) => {
+  patchMessage(auth, '(deleted)', message.id);
```
Comment (borisyankov) [inline src/message/messageActionSheet.js]: I suspect you might have to use '' as the value.
@timabbott is that how we want to delete a message?

Comment (saketkumar95) [inline src/message/messageActionSheet.js]: `patchMessage(auth, '' , message.id)` 
This too worked. I just thought not to leave the empty strings.

File: src/i18n/translations/messages_en.json (hunk 1 above)
Comment (saketkumar95) [inline src/i18n/translations/messages_en.json]: Done! 👍

File: src/message/messageActionSheet.js (hunk 2 above)
Comment (borisyankov) [inline src/message/messageActionSheet.js]: Why not? That is exactly the spirit of deleting a message.
I still wonder if @timabbot will suggest another way to delete a message. Maybe a deleteMessage() with DELETE HTTP method.

Comment (timabbott) [inline src/message/messageActionSheet.js]: Based on reading the backend code, I think we should be setting the content to `""`; the string `(deleted)` is what the backend replaces that with now, but we may change the markup in the future.

As a sidenote, we do support organization administrators completely deleting a message now, but that's a lower-priority feature that I don't think is worth working on for mobile yet.

Comment (borisyankov) [inline src/message/messageActionSheet.js]: Cool. @saketkumar95 please make sure that you are sending an empty string. Besides that, the PR is ready to merge and I'll do so after the change.

Comment (saketkumar95) [inline src/message/messageActionSheet.js]: @borisyankov Done! 👍

**Pull request merged**